code-execution-engines/langchain4j-code-execution-engine-v8/src/main/java/dev/langchain4j/agent/tool/v8/V8JavaScriptExecutionTool.java

```diff
@@ -0,0 +1,18 @@
+package dev.langchain4j.agent.tool.v8;
+
+
+import dev.langchain4j.agent.tool.P;
+import dev.langchain4j.agent.tool.Tool;
+import dev.langchain4j.code.CodeExecutionEngine;
+import dev.langchain4j.code.v8.V8JavaScriptExecutionEngine;
+
+public class V8JavaScriptExecutionTool {
+
+    private final CodeExecutionEngine engine = new V8JavaScriptExecutionEngine();
```

Comment:
QQ: How expensive is to create a new engine? Should it be reused and injected as a dependency or be a singleton with: `V8JavaScriptExecutionEngine().getInstance()` ?